File: Algorithms/Maths/Java/Leaders_in_Array_1.java
import java.util.Scanner;

public class Leaders_in_Array_1 {
    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        int n = sc.nextInt();

        int arr[] = new int[n], k = 0;

        System.out.print("Enter the numbers in the array: ");
        for(int i = 0; i < n; i++){
            arr[i] = sc.nextInt();
        }

        boolean isleader = true;
        int leaders[] = new int[n];

        for(int i = 0; i < n; i++){
            isleader = true;
            for(int j = (i + 1); j < n; j++){
                if (arr[i] < arr[j]) {
                    isleader = false;
                    break;
                }
            }
            if(isleader){
                leaders[k] = arr[i];
                k++;
            }
        }

        System.out.print("The Leaders in the array: ");
        for(int i = 0; i < k; i++){
            if(i == (k - 1)){
                System.out.print(leaders[i] + " ");
            }else{
                System.out.print(leaders[i] + ", ");
            }
        }

        sc.close();
    }
}

File: Algorithms/Maths/C/Leaders_in_Array_1.c
#include<stdio.h>
#include<stdbool.h>

int main(){

    int n, k = 0;
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int array[n];

    printf("Enter the numbers in the array: ");
    for(int i = 0; i < n; i++){
        scanf("%d", &array[i]);
    }

    bool isLeader = true;
    int leaders[n];

    for(int i = 0; i < n; i++){
        isLeader = true;
        for(int j = (i + 1); j < n; j++){
            if(array[i] < array[j]){
                isLeader = false;
                break;
            }
        }
        if(isLeader == true){
            leaders[k] = array[i];
            k++;
        }
    }

    printf("The Leaders in the array: ");
    for(int i = 0; i < k; i++){
        if(i == (k - 1)){
            printf("%d ", leaders[i]);
        }else{
            printf("%d, ", leaders[i]);
        }
    }

    return 0;
}